app/js/enhancedDate.js

```diff
@@ -1,5 +1,97 @@
-/* 
+/*
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ *
+ * excellent explaination of module method http://toddmotto.com/mastering-the-module-pattern/
+ *
+ *◦ setDate: Stores the passed-in date. Takes milliseconds after epoch or a date object. Do some
+ *  type-checking to make sure you have a valid type. If no value is passed, default to now.
+ *◦ getDate: Returns either milliseconds after epoch by default or a date object if true is passed as an
+ *  argument.
+ *◦ getDayName: Returns the full day name as a string, e.g. 'Monday', 'Tuesday', etc.
+ *◦ getMonthName: Return the full month name as a string, e.g. 'January', 'February', etc.
+ *◦ isFuture: Return boolean true if date is in the future (from when method is called)
+ *◦ isToday: Return boolean true if the stored date is 'today'.
+ *
+ *
+ *  JavaScript 300 Spring 2015
+ *  Ted Morrill-McClure
+ */
 
+var enhancedDate = (function() {
+    'use strict';
+
+    //create object
+    var myObject = {};
+
+    myObject.setDate = function(passedDate) {
+        var now;
+
+        if (passedDate) {
+            if (passedDate instanceof Date) {
+                now = passedDate;
```

Comment:
This works, but one thing to keep in mind is that an object will be passed in by reference, so if someone alters the original Date object passed in from outside the enhancedDate module, it will also modify the internally stored Date reference. A way to solve this is to simply instantiate a new Date object, passing in the Date object as an argument to the Date constructor - basically the same thing you do if they passed in milliseconds.